### ISSUE TITLE
Idea for object definition schemas

### ISSUE THREAD

**Issue by atdiff** [swagger-api/swagger-parser#175]
I think it would be useful to be able to retrieve the schema for a response (combined with the definitions so as to have a complete picture) and essentially have a simple JSON Schema for a response.  This would make it easier to then verify the API's JSON payload as well as other things.

- Example of a simple JSON schema that theoretically could be obtained
```
{
  "type": "array",
  "items": {
    "$ref": "#/definitions/state"
  },

  "definitions": {
    "state": {
      "properties": {
        "id": {
          "type": "integer"
        },
        "code": {
          "type": "string"
        },
        "abbreviation": {
          "type": "string"
        },
        "name": {
          "type": "string"
        }
      },
      "required": [
        "id",
        "code",
        "abbreviation",
        "name"
      ]
    }
  }
}
```

At a minimum, it'd be nice if there were convenience methods to make it easier to retrieve the schema from the Swagger object.  The following are two ways that it can be done currently which makes the code a bit wordy and also doesn't supply the most useful object - a Property:

```
// Unfortunately have to know that it's a getGet or do a little more finagling (example 2)
- Example 1
swaggerDeserializationResult.getSwagger().getPaths().get("/states").getGet().getResponses().getOrDefault("200", null).getSchema()

- Example 2
    String currentUri = ((HttpServletRequest)req).getRequestURI();
    HttpMethod currentMethod = HttpMethod.valueOf(((HttpServletRequest)req).getMethod());
    Path currentPath = swagger.getPath(currentUri);
    if(currentPath != null) {
    	Operation currentOperation = currentPath.getOperationMap().get(currentMethod);
    	currentOperation.toString();
    }
```

Note: this isn't a condemnation of the code as it could mean that I'm not using it in the intended manner.  If there's an alternative/better way to using/retrieving the schema, let me know.  Or a better way to achieve the result mentioned at the beginning.

@fehguy Thoughts?  Thanks in advance.

**Comment by fehguy** [comment]
This is expensive to do, which is why it hasn't been put in the parser logic.  There is an implementation here:

https://github.com/swagger-api/swagger-inflector/blob/68aa5d397331b69d691802c6fbf5a6c8fe5fd474/src/main/java/io/swagger/inflector/utils/ResolverUtil.java#L23
